src/control/security/grpc_authorization.go

```diff
@@ -0,0 +1,92 @@
+//
+// (C) Copyright 2019 Intel Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// GOVERNMENT LICENSE RIGHTS-OPEN SOURCE SOFTWARE
+// The Government's rights to use, modify, reproduce, release, perform, display,
+// or disclose this software are subject to the terms of the Apache License as
+// provided in Contract No. 8F-30005.
+// Any reproduction of computer software, computer software documentation, or
+// portions thereof marked with this legend must also reproduce the markings.
+//
+
+package security
+
+import "strings"
+
+// Component represents the DAOS component being granted authorization.
+type Component int
+
+const (
+	ComponentUndefined Component = iota
+	ComponentAdmin
+	ComponentAgent
+	ComponentServer
+)
+
+func (c Component) String() string {
+	return [...]string{"undefined", "admin", "agent", "server"}[c]
+}
+
+// methodAuthorizations is the map for checking which components are authorized to make the specific method call.
+var methodAuthorizations = map[string]Component{
```

Comment:
Is it true that only one component will be authorized for each method? Wondering if this should be a slice instead of a single Component.

Reply:
As of right now it is only one component. I don't have a usecase at the moment for two components needing to access the same method. If we ever do allow that we really should log which component the call is coming from then.

Reply:
OK. Should be fairly trivial to change it if a case like that eventually comes around.